packages/plugins/flyDrive/src/internal/plugin.ts

```diff
@@ -0,0 +1,56 @@
+import type { OberonPlugin } from "@oberoncms/core"
+
+import { Disk } from "flydrive"
+import { FSDriver } from "flydrive/drivers/fs"
+import { GCSDriver } from "flydrive/drivers/gcs"
+import { S3Driver } from "flydrive/drivers/s3"
+
+import { name, version } from "../../package.json" with { type: "json" }
+import { initRouteHandler } from "./disk-handlers"
+
+type CloudFlyDriver = S3Driver | GCSDriver | FSDriver
+const getDriverActions = (driver: CloudFlyDriver) => {
+  const disk = new Disk(driver)
+  if (driver instanceof FSDriver) {
+    return Object.assign(disk, {
+      type: "disk",
+    })
+  } else if (driver instanceof S3Driver) {
+    return Object.assign(disk, {
+      type: "s3",
+    })
+  } else if (driver instanceof GCSDriver) {
+    return Object.assign(disk, {
+      type: "gcs",
+    })
+  }
```

Comment:
nitpick: the "else" below are unnecessary and typescript plays better with spread assignment than Object.assign:
```suggestion
  if (driver instanceof FSDriver) {
    return {
      ...disk,
      type: "disk",
    }
  }
  if (driver instanceof S3Driver) {
    return {
      ...disk,
      type: "s3",
    }
  }
  if (driver instanceof GCSDriver) {
    return {
      ...disk,
      type: "gcs",
    }
  }
```